sklearn/manifold/spectral_embedding_.py

```diff
@@ -287,7 +291,8 @@ def spectral_embedding(adjacency, n_components=8, eigen_solver=None,
             embedding = diffusion_map.T[:n_components] * dd
         else:
             # lobpcg needs native floats
-            laplacian = laplacian.astype(np.float)
```

Comment:
We can actually remove that check as it's already done before entering the if / else blocks.